cube/src/main/java/zone/cogni/asquare/cube/index/ElasticsearchIndexingService.java

```diff
@@ -0,0 +1,604 @@
+    indexMetadata.setCollections(calculateCollectionsMetadata(indexMetadata));
+    return indexMetadata;
+  }
+
+  private Resource calculateElasticSettingsResource(String indexName) {
+    return getResource("/" + indexName + "/elastic-settings.json");
+  }
+
+  private List<CollectionMetadata> calculateCollectionsMetadata(IndexMetadata indexMetadata) {
+    Resource[] resources = getResources("/" + indexMetadata.getName() + "/**/*");
+    List<String> localPaths = getLocalPaths(resources);
+    return localPaths.stream()
+                     .map(this::stripSlashAtFront)
+                     .map(path -> StringUtils.substringAfter(path, indexMetadata.getName() + "/")) // <- strip index
+                     .filter(path -> path.contains("/")) // <- make sure it's a folder
+                     .map(path -> StringUtils.substringBefore(path, "/")) // <- take folder name
+                     .distinct()
+                     .sorted()
+                     .map(collectionName -> calculateCollectionMetadata(indexMetadata, collectionName))
+                     .collect(Collectors.toList());
+  }
+
+  private CollectionMetadata calculateCollectionMetadata(IndexMetadata indexMetadata, String collection) {
+    String index = indexMetadata.getName();
+
+    CollectionMetadata result = new CollectionMetadata();
+    result.setName(collection);
+    result.setSelectQueryResources(getCollectionSelectQueries(index, collection));
+    result.setConstructQueryResources(getCollectionConstructQueries(index, collection));
+    result.setFacetQueryResources(getFacetQueries(index, collection));
+    return result;
+  }
+
+  private List<Resource> getCollectionSelectQueries(String index, String collection) {
+    return Arrays.asList(getResources("/" + index + "/" + collection + "/select-*.*"));
+  }
+
+  private List<Resource> getCollectionConstructQueries(String index, String collection) {
+    return Arrays.asList(getResources("/" + index + "/" + collection + "/construct-*.*"));
+  }
+
+  @PostConstruct
+  public void validate() {
+    this.indexesMetadata = calculateIndexesMetadata();
+
+    log.info("'{}' index service", configurationClasspath);
+
+    List<String> indexNames = getIndexNames();
+    log.info("{} index count: {}", indent, indexNames.size());
+
+    if (indexNames.isEmpty()) {
+      log.error("{} no valid indexes configured", indent);
+    }
+
+    if (!indexNames.isEmpty())
+      log.info("{} valid indexes:   {}", indent, String.join(", ", indexNames));
+    if (!getInvalidIndexNames().isEmpty())
+      log.info("{} invalid indexes: {}", indent, String.join(", ", getInvalidIndexNames()));
+
+    indexesMetadata.forEach(this::validateIndex);
+  }
+
+  private void validateIndex(IndexMetadata indexMetadata) {
+    // valid or not
+    if (indexMetadata.isValid())
+      log.info("  '{}' index is valid", indexMetadata.getName());
+    else
+      log.error("  '{}' index NOT is valid", indexMetadata.getName());
```

Comment:
if we would do this, I think it would be better look at spring boot options here and not a DIY solution. for now I would consider this a new feature and create an issue.

Reply:
on the other hand: bad configuration also just crashes. will add it here.